examples/cvode/serial/cvRocket_dns.c

```diff
@@ -0,0 +1,415 @@
+  if (check_retval(&retval, "CVodeInit", 1))
+    return (1);
+
+  /* Call CVodeSVtolerances to specify the scalar relative tolerance and vector absolute tolerances */
+  retval = CVodeSVtolerances(cvode_mem, reltol, abstol);
+  if (check_retval(&retval, "CVodeSVtolerances", 1))
+    return (1);
+
+  /* Provide sunbooleantype engine_on as user data for use in f and g routines */
+  retval = CVodeSetUserData(cvode_mem, &engine_on);
+  if (check_retval((void*)&retval, "CVodeSetUserData", 1))
+    return (1);
+
+  /* Call CVodeRootInit to specify the root function g with 2 components */
+  retval = CVodeRootInit(cvode_mem, 2, g);
+  if (check_retval(&retval, "CVodeRootInit", 1))
+    return (1);
+
+  /* Create dense SUNMatrix for use in linear solves */
+  A = SUNDenseMatrix(NEQ, NEQ, sunctx);
+  if (check_retval((void*)A, "SUNDenseMatrix", 0))
+    return (1);
+
+  /* Create dense SUNLinearSolver object for use by CVode */
+  LS = SUNLinSol_Dense(y, A, sunctx);
+  if (check_retval((void*)LS, "SUNLinSol_Dense", 0))
+    return (1);
+
+  /* Call CVodeSetLinearSolver to attach the matrix and linear solver to CVode */
+  retval = CVodeSetLinearSolver(cvode_mem, LS, A);
+  if (check_retval(&retval, "CVodeSetLinearSolver", 1))
+    return (1);
+
+  /* Set the user-supplied Jacobian routine Jac */
+  retval = CVodeSetJacFn(cvode_mem, Jac);
+  if (check_retval(&retval, "CVodeSetJacFn", 1))
+    return (1);
+
+  /* In loop, call CVode, print results, check for root stops, and test for error.  On the first
+     root return, restart with engine turned off. Break out of loop when NOUT preset output times
+     have been reached, or when the returned value of H is negative.  */
+  printf(" \nAccelerating rocket problem\n\n");
+
+  iout      = 0;
+  tout      = T1;
+  engine_on = SUNTRUE;
+  numroot   = 2;
+  while (1) {
+    retval = CVode(cvode_mem, tout, y, &t, CV_NORMAL);
+    PrintOutput(t, NV_Ith_S(y, 0), NV_Ith_S(y, 1));
+
+    if (engine_on && (retval == CV_ROOT_RETURN)) { /* engine cutoff */
+      retvalr = CVodeGetRootInfo(cvode_mem, rootsfound);
+      if (check_retval(&retvalr, "CVodeGetRootInfo", 1))
+        return (1);
+      PrintRootInfo(rootsfound[0], rootsfound[1], numroot);
+      engine_on = SUNFALSE;
+      numroot   = 1;
+      /* Call CVodeRootInit to specify the root function g with 1 component */
+      retval = CVodeRootInit(cvode_mem, 1, g);
+      if (check_retval(&retval, "CVodeRootInit", 1))
+        return (1);
+      /* Reinitialize the solver with current t and y values. */
+      retval = CVodeReInit(cvode_mem, t, y);
+      if (check_retval((void*)&retval, "CVodeReInit", 1))
+        return (1);
+    } else if ((!engine_on) && (retval == CV_ROOT_RETURN)) { /* max.  height */
+      retvalr = CVodeGetRootInfo(cvode_mem, rootsfound);
+      if (check_retval(&retvalr, "CVodeGetRootInfo", 1))
+        return (1);
+      PrintRootInfo(rootsfound[0], rootsfound[1], numroot);
+    }
+
+    if (check_retval(&retval, "CVode", 1))
```

Comment:
Since this is catching an error return from CVODE I think it should go right after the call to `CVode`